lldb/source/Plugins/TypeSystem/Swift/SwiftASTContext.cpp

```diff
@@ -1573,7 +1573,21 @@ bool ShouldUnique(StringRef arg) {
 
 // static
 void SwiftASTContext::AddExtraClangArgs(const std::vector<std::string> &source,
-                                        std::vector<std::string> &dest) {
+                                        std::vector<std::string> &dest,
+                                        bool cc1) {
+  // FIXME: Support for cc1 flags isn't complete.  The uniquing
```

Comment:
The flags from a normal build (that is planned by swift dependency scanner), this cc1 list is absolutely deterministic and have a stable ordering already but probably lldb should not rely on this fact.

What really should happen is for lldb to create a `clang::CompilerInvocation` and parse the args then it can be systematically collect/modify the flag values. I am ok to leave that for future improvement

Reply:
> What really should happen is for lldb to create a clang::CompilerInvocation and parse the args then it can be systematically collect/modify the flag values. I am ok to leave that for future improvement

That might work, and if it does that would be a better implementation. It's possible that some modifications have to happen before parsing though, because the unmodified arguments wouldn't pass validation.

Reply:
Oh, you mean some old cc1 args that are no longer valid for current clang version? That would be tricky (even though we expect a version matching lldb for debugging swift).

Reply:
That case is hopeless for explicit modules anyway (the pcms would be incompatible), and I wouldn't expect this mode to be used for anything but explicit modules.

Reply:
> That might work, and if it does that would be a better implementation. It's possible that some modifications have to happen before parsing though, because the unmodified arguments wouldn't pass validation.

I was thinking about a hypothetical scenario of the driver complaining if a path doesn't exist (which it probably doesn't do)